app/scripts/modules/cloudfoundry/src/pipeline/config/validation/instanceSize.validator.ts

```diff
@@ -35,12 +35,21 @@ export class CfInstanceSizeFieldValidator implements IStageOrTriggerValidator {
 
   protected validationMessage(validationConfig: IInstanceFieldSizeValidationConfig, pipeline: IPipeline): string {
     const fieldLabel: string = this.printableFieldLabel(validationConfig);
-    const min: any = has(validationConfig, 'min') ? get(validationConfig, 'min') : 'NA';
-    const max: any = has(validationConfig, 'max') ? get(validationConfig, 'max') : 'NA';
-    return (
-      validationConfig.message ||
-      `<strong>${fieldLabel}</strong> should be between ${min} and ${max} in ${pipeline.name}`
-    );
+    const hasMin = has(validationConfig, 'min');
+    const min: any = hasMin ? get(validationConfig, 'min') : 'NA';
+    const hasMax = has(validationConfig, 'max');
+    const max: any = hasMax ? get(validationConfig, 'max') : 'NA';
+    let message = ``;
+    if (hasMin) {
+      if (hasMax) {
+        message = `<strong>${fieldLabel}</strong> should be between ${min} and ${max} in ${pipeline.name}.`;
+      } else {
+        message = `<strong>${fieldLabel}</strong> should be at least ${min} in ${pipeline.name}.`;
+      }
+    } else if (hasMax) {
+      message = `<strong>${fieldLabel}</strong> should be no more than ${max} in ${pipeline.name}.`;
+    }
```

Comment:
I think that there are two separate issues in your comment.  We have addressed the implied exclusivity of the comment by rewording it.  However, regardless of the content of the three cases, the set of 'if' statements must be traversed regardless of which case we're handling.  We feel that there's no real computational gain by factoring-out the beginnings and ends of the messages, and the full text of each string is perhaps more legible (and thusly more maintainable in the future) with the full sentences being present.

Reply:
The wording looks correct to me now.
The message similarity is a less serious thing because it still produces correct output, but I'd like to be clear about why I mentioned it. Refraining from duplicating the common part of the message was not suggested as a performance optimization. It is generally easier to reason about conditionals when they contain only the differing logic, and it is going to be less prone to inconsistent changes with less repetition. It's not much of a leap for the reader to discern the final string.

Reply:
In general, I believe what you say about repetition is true about code, but I'm not convinced this is true about error messages and providing clarity in natural language.  In the case of messages, don't you think that it's clearer to someone who is maintaining the code what message should actually be expected? We're not simply interpolating one or two labels, but rather we are changing the structure of the sentence.